### DISCUSSION TITLE
Read EdgeTX Radio Telemetry using the in-progress Serial Transmitter interface?

### DISCUSSION THREAD

**Discussion by matt-nolan11** [ZZ-Cat/CRSFforArduino#118] [Q&A] [answered]
Hi again,

I'd like to be able to read the telemetry coming from my Radiomaster TX16S using a microcontroller. The TX16 has 2 AUX ports at the bottom with RX and TX lines, and they can be set to "Mirror Telemetry" in the hardware settings of the radio.

The radio's telemetry can be mirrored to the USB-VCP port as well, and I was successfully able to receive and parse it on a PC using this [Python example script](https://github.com/crsf-wg/crsf/wiki/Python-Parser) (using CRSFforArduino to generate the telemetry values on the receiver side!). However, for reasons that are pretty specific to my project, I'd like to be able to parse the telemetry using a microcontroller.

This may be outside the scope of #88, but do you think it would be possible to read the ELRS telemetry being mirrored to the radio's AUX port using the Serial Transmitter interface?

**Comment by ZZ-Cat** [comment, marked as answer]
Good question, bud.

I am yet to try out the telemetry mirroring feature myself. So, this would be a golden opportunity for me to check it out.

Currently, there isn't any implementation to read pure telemetry with CRSF for Arduino... yet.  
However, if you would like this to be added in, you're more than welcome to open a [Feature Request](https://github.com/ZZ-Cat/CRSFforArduino/issues/new?assignees=&labels=%E2%9C%A8%EF%B8%8F+Enhancement+%E2%9C%A8%EF%B8%8F%2CPending+%F0%9F%93%8C&projects=&template=feature-request.yml), and it will make it into the upcoming Version 1.1.0 release.

> I was successfully able to receive and parse it on a PC using this Python example script

That's funny. 😆
'Cause I was dicking around with Python over the last two nights, and I practically replicated the RC data parsing from CRSF for Arduino's Serial Receiver Interface with some promising results.
